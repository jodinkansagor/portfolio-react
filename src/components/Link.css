.linkSection {
  img {
    width: 40px;
    height: 40px;
    margin-right: 10px;
  }

  h2 {
    display: none;
  }
  
}
