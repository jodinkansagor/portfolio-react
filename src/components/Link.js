import React from 'react';
import PropTypes from 'prop-types';
import styles from '../components/Link.css';

const Link = ({ url, linkImg, linkName }) => {
  return (
    <section className={styles.linkSection}>
      <a href={url}><img src={linkImg} /></a>
      <h2>{linkName}</h2>
    </section>
  );
};

Link.propTypes = {
  url: PropTypes.string.isRequired,
  linkImg: PropTypes.string.isRequired,
  linkName: PropTypes.string.isRequired
}

export default Link;
