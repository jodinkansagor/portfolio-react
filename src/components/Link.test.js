import React from 'react';
import { shallow } from 'enzyme';
import Link from './Link';

describe('Link component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<Link link={'link'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
