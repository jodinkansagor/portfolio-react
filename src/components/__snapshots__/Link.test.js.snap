// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component matches a snapshot 1`] = `
<section
  className="linkSection"
>
  <a
    target="_blank"
  >
    <img />
    <h2 />
  </a>
</section>
`;
